src/app/modules/analytics-config/components/add-subtable-dialog/add-subtable-dialog.component.ts

```diff
@@ -0,0 +1,134 @@
+import { CollectionField } from '@services/collection-service/models';
+import { NUMERIC_OR_DATE_TYPES } from '@services/collection-service/tools';
+import { SelectFormControl } from '@shared-models/config-form';
+import { Metric } from 'arlas-api';
+
+@Component({
+  selector: 'arlas-add-subtable-dialog',
+  templateUrl: './add-subtable-dialog.component.html',
+  styleUrls: ['./add-subtable-dialog.component.scss']
+})
+export class AddSubtableDialogComponent implements OnInit {
+  public formGroup: SubTableFormGroup;
+  public defaultKey: string;
+
+  @ViewChild('columnTable', { static: true }) public columnTable;
+
+  public dragDisabled = true;
+
+  public displayedColumns: string[] = ['action', 'metric', 'field'];
+  public constructor(
+    @Inject(MAT_DIALOG_DATA) public dialogData: any,
+    public dialogRef: MatDialogRef<AddSubtableDialogComponent>,
+    private metricsTableFormBuilder: MetricsTableFormBuilderService,
+    private collectionService: CollectionService
+  ) { }
+
+  public ngOnInit(): void {
+    const formBuilder = this.metricsTableFormBuilder;
+    if (!!this.dialogData.subTable) {
+      this.formGroup = this.dialogData.subTable;
+      (this.formGroup.get('columns') as FormArray).controls.forEach(c => {
+        this.initMetricCollectField(c as SubTableColumnFormGroup );
+      });
+    } else {
+      this.formGroup = formBuilder.buildSubTable(this.dialogData.collection);
+    }
+    this.defaultKey = formBuilder.defaultKey;
+    if(this.formGroup){
+      this.formGroup.get('collection').valueChanges.subscribe(v => {
+        this.columns.clear();
+        this.columnTable.renderRows();
+      });
+    }
+  }
+
+  public add(): void {
+    this.formGroup.markAllAsTouched();
+    if (this.formGroup.valid) {
+      this.dialogRef.close(this.formGroup);
+    }
+  }
+
+  public addColumn() {
+    const subTableColumn = this.metricsTableFormBuilder.buildSubTableColumn(this.collection);
+    this.initMetricCollectField(subTableColumn);
+    this.columns.push(subTableColumn);
+    this.columnTable.renderRows();
+  }
+
+  private initMetricCollectField(subTableColumn: SubTableColumnFormGroup) {
+    const control: SelectFormControl = subTableColumn.get('metricCollectField') as SelectFormControl;
+    control.disable();
+    subTableColumn.get('metricCollectFunction').valueChanges.subscribe(v => {
+      if (v && v !== 'count') {
+        control.enable();
+        const filterCallback = (field: CollectionField) => v === Metric.CollectFctEnum.CARDINALITY ?
+          field : NUMERIC_OR_DATE_TYPES.indexOf(field.type) >= 0;
+        const sub = this.collectionService.getCollectionFields(this.collection).subscribe(
+          fields => {
+            control.setSyncOptions(
+              fields
+                .filter(filterCallback)
+                .sort((a, b) => a.name.localeCompare(b.name))
+                .map(f => ({ value: f.name, label: f.name, enabled: f.indexed })));
+            sub.unsubscribe();
```

Comment:
If the method is an HTTP call it is not necessary to unsubscribe once the call has completed. I read a bit about it, and people advise to still have the takeUntil on http requests to avoid processing a query's response when the component is destroyed, to ensure coherence